content/3_Silver/Sorting_Custom.mdx

```diff
@@ -48,626 +46,461 @@ After sorting, it should look like
-[const Edge&](https://stackoverflow.com/questions/11805322/why-should-i-use-const-t-instead-of-const-t-or-t)):
+Let's sort those four edges again with this new method:
 
 ```cpp
-#include <bits/stdc++.h>
+#include <algorithm>
+#include <iostream>
+#include <vector>
+
 using namespace std;
 
 struct Edge {
-	int a, b, w;
-	bool operator<(const Edge &y) { return w < y.w; }
+	int a, b;
+	int width;
+
+	// see https://stackoverflow.com/a/11805395/12128483 for why we use const Edge&
+	bool operator<(const Edge &y) { return width < y.width; }
 };
 
 int main() {
-	int M = 4;
-	vector<Edge> v;
-	for (int i = 0; i < M; ++i) {
-		int a, b, w;
-		cin >> a >> b >> w;
-		v.push_back({a, b, w});
-	}
-	sort(begin(v), end(v));
-	for (Edge e : v) cout << e.a << " " << e.b << " " << e.w << "\n";
+	constexpr int edge_num = 4;
+
+	vector<Edge> edges(edge_num);
+	for (Edge &e : edges) { cin >> e.a >> e.b >> e.width; }
+	sort(edges.begin(), edges.end());
+
+	for (const Edge &e : edges) { printf("(%i, %i): %i\n", e.a, e.b, e.width); }
 }
 ```
 
-We can also overload the operator outside of the class:
+It's also possible to define `operator<` outside of the class:
 
 ```cpp
 struct Edge {
-	int a, b, w;
+	int a, b;
+	int width;
 };
-bool operator<(const Edge &x, const Edge &y) { return x.w < y.w; }
+
+bool operator<(const Edge &x, const Edge &y) { return x.width < y.width; }
 ```
 
 or within it using
-[friend](https://www.geeksforgeeks.org/friend-class-function-cpp/):
+[`friend`](https://en.cppreference.com/w/cpp/language/friend):
 
 ```cpp
 struct Edge {
-	int a, b, w;
-	friend bool operator<(const Edge &x, const Edge &y) { return x.w < y.w; }
+	int a, b;
+	int width;
+
+	friend bool operator<(const Edge &x, const Edge &y) { return x.width < y.width; }
 };
 ```
 
-### Method 2 - Comparison Function
+### Comparison Function
 
-This works for both objects and primitives, and you can declare many different
-comparators for the same object.
+We can also pass the comparator as a
+[lambda](https://en.cppreference.com/w/cpp/language/lambda)
+directly into `std::sort`.
+This has the advantage of working with primitives as well as classes.
 
 ```cpp
-#include <bits/stdc++.h>
+#include <algorithm>
+#include <iostream>
+#include <vector>
+
 using namespace std;
 
 struct Edge {
-	int a, b, w;
+	int a, b;
+	int width;
 };
 
-bool cmp(const Edge &x, const Edge &y) { return x.w < y.w; }
-
 int main() {
-	int M = 4;
-	vector<Edge> v;
-	for (int i = 0; i < M; ++i) {
-		int a, b, w;
-		cin >> a >> b >> w;
-		v.push_back({a, b, w});
-	}
-	sort(begin(v), end(v), cmp);
-	for (Edge e : v) cout << e.a << " " << e.b << " " << e.w << "\n";
-}
-```
+	constexpr int edge_num = 4;
 
-We can also use [lambda expressions](/general/lambda) in C++11 or above:
+	vector<Edge> edges(edge_num);
+	for (Edge &e : edges) { cin >> e.a >> e.b >> e.width; }
+	sort(edges.begin(), edges.end(),
+	     [&](const Edge &x, const Edge &y) { return x.width < y.width; });
```

Comment:
do we need an `&` capture? 👀

Reply:
bro probably just typed it outta habit lol

Reply:
i mean it's always nice to not copy stuff
if you insist i can change it to just `Edge`

Reply:
no sans i dont think yk what he means by & capture

basically, lambdas in their declaration can capture stuff

e.g.

```cpp
[capture_stuff](params) -> return_type {}
```

`[&]` means you capture everything in the local scope. here we don't need to capture everything, so just do `[]`

Reply:
(although tbf i doubt there's ever a meaningful performance difference between the two)

Reply:
surely it doesn't matter :clueless:

Reply:
regardless i removed it